### DISCUSSION TITLE
Redis Pool & Other Java Lib

### DISCUSSION THREAD

**Discussion by nguyennhatdung** [quarkusio/quarkus#29629] [Q&A] [answered]
Hi!

I'm new to quarkus, I'm trying convert my project to quarkus

i just wonder is there any way to use my old lib :  Jedis , JedisPool, org.apache.httpcomponents.client5 ?

seem Quarkus doesn't have Pool for Redis

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
Quarkus provides multiple Redis APIs from low level to high level, but it does not use JEdis. 
You can use Jedis in JVM mode, but it won't work in native. I would recommend switching to the Quarkus Redis API. See: 

- https://quarkus.io/guides/redis
- https://quarkus.io/guides/redis-reference
- https://quarkus.io/blog/redis-api-intro/

And, of course, it has a connection pool.

**Comment by nguyennhatdung** [comment]
Thanks for the info , i just search but cannot find anything about pool in Quarkus Redis Document. 

Normally i use JedisPool in java, what is the equivalent in Quarkus ?

**Reply by cescoffier** [reply]
You just inject the client, it will use the pool accordingly. So:

@Inject RedisDataSource ds;

Or

@Inject Redis redis; (low level)

**Reply by nguyennhatdung** [reply]
many thx, i'll try that

**Reply by daloji** [reply]
Salut,
Our application is an REST service that uses a RedisClient to interact with our Redis Server
each http request to our REST service can generate about ten connections to Redis.

To handle so much connection, we have an object pool pattern to handle Redis clients.
otherwise we have the following exception 
  
 ```io.vertx.core.http.ConnectionPoolTooBusyException: Connection pool reached max wait queue size of 24```

for this we use GenericObjectPool (Apache Commons Object Pooling Library ) to manage the thread pool. it works well in quarkus version 2.9.2

we are updating our quarkus (2.16.2)  (RedisClient) is deprecated so we are using RedisDatasource.
But unfortunately we can't use our object pool pattern anymore 
we have again the exception
```io.vertx.core.http.ConnectionPoolTooBusyException: Connection pool reached max wait queue size of 24 **```
because  1 single client cannot manage so many connections.
I tried with Jedis and it works fine

I reopen this discussion because I can't find a solution with RedisDatasource in synchronous mode

**Reply by cescoffier** [reply, marked as answer]
You need to configure the number of connections of the Redis client using:

- quarkus.redis.max-pool-size: number of connections
- quarkus.redis-max-pool-waiting: max waiting queue size

**Reply by daloji** [reply]
thank you.
it works
